src/utils/Printer.cpp

```diff
@@ -154,4 +147,35 @@ int Printer::setPrecision(int i) {
     return oldPrec;
 }
 
+//parse a string and returns the nth integer number
+int Printer::getVal(char* line, int n){
+    char* p = line;
+    int len = 0;
+    for(int i=0;i<n-1;i++){
+        //jump over n-1 first numbers
+        while (*p <'0' || *p > '9') p++;
+        while (*p >='0' && *p <= '9') p++;
+    }
+    while (*p <'0' || *p > '9') p++;
+    char* s = p;
+    while (*s >='0' && *s <= '9'){ s++; line[len] = p[len]; len++;}
+    p[len] = 0;
+    return atoi(p);
+}
+
+/** Prints (and returns) the current memory usage of this process
+ */
+int Printer::printMem(char* txt){
+    FILE* file = fopen("/proc/self/statm", "r");
```

Comment:
Is this system independent?

Reply:
I do not think so. Or maybe it must be linux. I must say I do not understand the details: it is not a real file, as it might look like!
"The proc filesystem is a pseudo-filesystem which provides an
       interface to kernel data structures.  It is commonly mounted at /proc. "

Reply:
Wrong/ambiguous answer before: I meant "I think this is system independent!"